File: Kattis/Java/NumberFun.java
package numberfun;

import java.util.Scanner;

public class NumberFun {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int testCaseAmount = input.nextInt();
        double a, b, c;
        for (int i = 0; i < testCaseAmount; i++) {
            a = input.nextDouble();
            b = input.nextDouble();
            c = input.nextDouble();
            
            if ((a + b) == c)
                System.out.println("Possible");
            else if ((a - b) == c || (b - a) == c)
                System.out.println("Possible");
            else if ((a / b) == c || (b / a) == c) 
                System.out.println("Possible");
            else if ((a * b) == c)
                System.out.println("Possible");
            else 
                System.out.println("Impossible");
        }
    }
    
}

File: Kattis/CPlusPlus/NumberFun.cpp
#include <iostream>

int main() {
    int testCaseAmount;
    double a, b, c;
    std::cin >> testCaseAmount;

    for (int i = 0; i < testCaseAmount; i++) {
        std::cin >> a >> b >> c;

        if ((a + b) == c) {
            std::cout << "Possible" << std::endl;
        } else if ((a - b) == c || (b - a) == c) {
            std::cout << "Possible" << std::endl;
        } else if ((a / b) == c || (b / a) == c) {
            std::cout << "Possible" << std::endl;
        } else if ((a * b) == c) {
            std::cout << "Possible" << std::endl;
        } else {
            std::cout << "Impossible" << std::endl;
        }
    }

    exit(EXIT_SUCCESS);
    std::cin.get();
}